### PR TITLE
feat(LiveLinkBio): add presence

### LINKED ISSUE #6085: LiveLinkBio | livelinkbio.com
### Website name

livelinkbio

### Website URL

https://livelinkbio.com/

### Website logo

https://cdn.discordapp.com/attachments/957097221210390640/960382757102551100/PnzJfuE3_400x400.png

### Prerequisites

- [ ] It is a paid service
- [ ] It displays NSFW content
- [ ] It is region restricted

### Description

basically I want it so that it shows up on the side of Discord to see how long you actually been on the website and the logo

### PR DESCRIPTION
**Pull request by DarkVillager**
**Describe the changes in this pull request:**
Resolves #6085
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>

</details>

### CODE REVIEW

File: websites/L/LiveLinkBio/presence.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+const presence = new Presence({
+	clientId: "966711989533544580"
+});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "logo"
+		},
+		browsingStamp = Math.floor(Date.now() / 1000),
```
Comment (EncryptedDev) [inline websites/L/LiveLinkBio/presence.ts:9]: This should be set outside the UpdateData event

File: websites/L/LiveLinkBio/presence.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+			presence.getSetting<boolean>("buttons")
+		]),
+		search = document.querySelector<HTMLInputElement>(
+			"#portals-root > div:nth-child(3) > div > div > div > div > div.css-1dbjc4n.r-1ro0kt6.r-16y2uox.r-1wbh5a2 > input"
+		);
+	let title: string;
+	if (privacy) presenceData.details = "Browsing...";
+	else if (document.location.hostname.includes("docs.livelinkbio.com")) {
+		presenceData.startTimestamp = browsingStamp;
+		if (search && search.value !== "") {
```
Comment (EncryptedDev) [inline websites/L/LiveLinkBio/presence.ts:21]: An empty string is falsy

File: websites/L/LiveLinkBio/presence.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		presenceData.startTimestamp = browsingStamp;
+		if (document.location.pathname == "/") presenceData.details = "Homepage";
+		else if (document.location.pathname.includes("affiliate")) {
+			presenceData.details = "Affiliate";
+			title = "Affiliate";
+		} else if (document.location.pathname.includes("dashboard")) {
+			presenceData.details = "Dashboard";
+			title = "Dashboard";
+		} else if (document.location.pathname.includes("link")) {
+			const link = document.querySelector("#link_url").textContent;
```
Comment (EncryptedDev) [inline websites/L/LiveLinkBio/presence.ts:46]: If you do this, you will not need L49 or L52
```suggestion
			presenceData.state = document.querySelector("#link_url").textContent;
```

File: websites/L/LiveLinkBio/presence.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+		} else if (active) {
+			title = active.textContent.trim();
+			presenceData.details = active.textContent.trim();
+		}
+	}
+	if (!privacy && buttons && document.location.pathname !== "/") {
+		if (!title) title = "Page";
+		presenceData.buttons = [
+			{
+				label: `View ${title}`,
```
Comment (EncryptedDev) [inline websites/L/LiveLinkBio/presence.ts:75]: Isn't there a possiblity this exceeds rate limit?

File: websites/L/LiveLinkBio/presence.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+				.querySelectorAll('[class="modal-title"]')[1]
+				.textContent.replace("Edit", "");
+			presenceData.details = `${document
+				.querySelectorAll('[class="modal-title"]')[1]
+				.textContent.replace("Edit", "")}:`;
+			presenceData.state =
+				document.querySelector<HTMLInputElement>("#update_name").value;
+		} else if (type && !active) {
+			title = type.textContent.trim();
+			presenceData.details = `All ${type.textContent.trim()}`;
```
Comment (EncryptedDev) [inline websites/L/LiveLinkBio/presence.ts:65]: `type.textContent.trim()` holds the same value as `title` above?

File: websites/L/LiveLinkBio/presence.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+		} else if (active) {
+			title = active.textContent.trim();
+			presenceData.details = active.textContent.trim();
```
Comment (EncryptedDev) [inline websites/L/LiveLinkBio/presence.ts:68]: `active.textContent.trim()` don't the same value as `title` above?

File: websites/L/LiveLinkBio/presence.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+				presenceData.details = "Editing Link:";
+				presenceData.state = link;
+			}
+		} else if (document.querySelector("body").className.includes("open")) {
+			title = document
+				.querySelectorAll('[class="modal-title"]')[1]
+				.textContent.replace("Edit", "");
+			presenceData.details = `${document
+				.querySelectorAll('[class="modal-title"]')[1]
+				.textContent.replace("Edit", "")}:`;
```
Comment (EncryptedDev) [inline websites/L/LiveLinkBio/presence.ts:60]: Ditto

File: websites/L/LiveLinkBio/presence.ts [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+			presenceData.details = "Searching For:";
+			presenceData.state = search.value;
+		} else {
+			presenceData.details = "Reading Docs:";
+			presenceData.state = document.querySelector(
+				"body > div.gitbook-root > div > div > div.css-1dbjc4n.r-1ro0kt6.r-16y2uox.r-1wbh5a2 > div.css-1dbjc4n.r-14lw9ot.r-13awgt0 > div > div > div.css-1dbjc4n.r-1ro0kt6.r-18u37iz.r-16y2uox.r-1wbh5a2 > div > div > div.css-1dbjc4n.r-eqz5dr.r-1ifxtd0.r-1ygmrgt > div.css-1dbjc4n.r-18u37iz > div.css-1dbjc4n.r-1ro0kt6.r-16y2uox.r-1wbh5a2 > div"
+			).textContent;
+		}
+	} else if (document.location.hostname.includes("status.livelinkbio.com"))
+		presenceData.details = "LiveLinkBio's Status";
```
Comment (EncryptedDev) [inline websites/L/LiveLinkBio/presence.ts:31]: "LiveLinkBio's" would be displayed as the presence name so this is unneeded?

**Review by EncryptedDev — COMMENTED** [review]
Hi, take a look at my review please :)

File: websites/L/LiveLinkBio/presence.ts [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+		} else if (document.location.pathname.includes("link")) {
+			presenceData.state = document.querySelector("#link_url").textContent;
+			if (document.location.pathname.endsWith("statistics"))
+				presenceData.details = "Statistics of Link:";
+			else presenceData.details = "Editing Link:";
+		} else if (document.querySelector("body").className.includes("open")) {
+			const modal = document.querySelectorAll('[class="modal-title"]')[1]
+				.textContent;
+			title = modal.replace("Edit", "");
+			presenceData.details = `${modal.replace("Edit", "")}:`;
```
Comment (EncryptedDev) [inline websites/L/LiveLinkBio/presence.ts:54]: Value of details is equal to title

File: websites/L/LiveLinkBio/presence.ts [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+			title = "Dashboard";
+		} else if (document.location.pathname.includes("link")) {
+			presenceData.state = document.querySelector("#link_url").textContent;
+			if (document.location.pathname.endsWith("statistics"))
+				presenceData.details = "Statistics of Link:";
+			else presenceData.details = "Editing Link:";
+		} else if (document.querySelector("body").className.includes("open")) {
+			const modal = document.querySelectorAll('[class="modal-title"]')[1]
+				.textContent;
+			title = modal.replace("Edit", "");
```
Comment (EncryptedDev) [inline websites/L/LiveLinkBio/presence.ts:53]:
```suggestion
			title = document.querySelectorAll('[class="modal-title"]')[1]
				.textContent.replace("Edit", "");
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/L/LiveLinkBio/dist/metadata.json [hunk 11]
```diff
@@ -0,0 +1,36 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "Dark_Ville",
+		"id": "638080361179512853"
+	},
+	"service": "LiveLinkBio",
+	"description": {
+		"en": "livelinkbio is an advanced URL Shortener Platform, that allows you to easily shorten links, target your audience based on their location or platform, and provide analytics insights for the shortened links.",
+		"nl": "livelinkbio is een geavanceerde URL Verkorter Platform, waar je makkelijk links kan verkorten, om jouw gekozen publiek te bereiken op basis van locatie en platform en het bied analytics aan voor de verkorte links."
```
Comment (EncryptedDev) [inline websites/L/LiveLinkBio/dist/metadata.json:10]: @Bas950

File: websites/L/LiveLinkBio/presence.ts [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+		]),
+		search = document.querySelector<HTMLInputElement>(
+			"#portals-root > div:nth-child(3) > div > div > div > div > div.css-1dbjc4n.r-1ro0kt6.r-16y2uox.r-1wbh5a2 > input"
+		);
+	let title: string;
+	if (privacy) presenceData.details = "Browsing...";
+	else if (document.location.hostname.includes("docs.livelinkbio.com")) {
+		presenceData.startTimestamp = browsingStamp;
+		if (search && !search.value) {
+			presenceData.details = "Searching For:";
```
Comment (EncryptedDev) [inline websites/L/LiveLinkBio/presence.ts:22]:
```suggestion
			presenceData.details = "Searching for:";
```

File: websites/L/LiveLinkBio/presence.ts [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("buttons")
+		]),
+		search = document.querySelector<HTMLInputElement>(
+			"#portals-root > div:nth-child(3) > div > div > div > div > div.css-1dbjc4n.r-1ro0kt6.r-16y2uox.r-1wbh5a2 > input"
+		);
+	let title: string;
+	if (privacy) presenceData.details = "Browsing...";
+	else if (document.location.hostname.includes("docs.livelinkbio.com")) {
+		presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/L/LiveLinkBio/presence.ts:20]: Set this in the presenceData obj instead

File: websites/L/LiveLinkBio/presence.ts [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+			title = active.textContent.trim();
+			presenceData.details = title;
+		}
+	}
+	if (!privacy && buttons && document.location.pathname !== "/") {
+		if (!title) title = "Page";
+		if (presenceData.buttons[0].label !== `View ${title}`) {
+			presenceData.buttons = [
+				{
+					label: `View ${title}`,
```
Comment (EncryptedDev) [inline websites/L/LiveLinkBio/presence.ts:70]: It isn't advised to use dynamic data in buttons

**Review by EncryptedDev — DISMISSED** [review]

File: websites/L/LiveLinkBio/presence.ts (hunk 14 above)
Comment (DarkVillager) [reply websites/L/LiveLinkBio/presence.ts:70]: okay

File: websites/L/LiveLinkBio/dist/metadata.json [hunk 15]
```diff
@@ -0,0 +1,36 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "Dark_Ville",
+		"id": "638080361179512853"
+	},
+	"service": "LiveLinkBio",
+	"description": {
+		"en": "livelinkbio is an advanced URL Shortener Platform, that allows you to easily shorten links, target your audience based on their location or platform, and provide analytics insights for the shortened links.",
```
Comment (ImRodry) [inline websites/L/LiveLinkBio/dist/metadata.json:9]:
```suggestion
		"en": "LiveLinkBio is an advanced URL Shortener Platform that allows you to easily shorten links, target your audience based on their location or platform, and provide insights on analytics for the shortened links.",
```

File: websites/L/LiveLinkBio/dist/metadata.json [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+		"nl": "livelinkbio is een geavanceerde URL Verkorter Platform, waar je makkelijk links kan verkorten, om jouw gekozen publiek te bereiken op basis van locatie en platform en het bied analytics aan voor de verkorte links."
+	},
+	"url": ["livelinkbio.com", "status.livelinkbio.com", "docs.livelinkbio.com"],
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/mqJJ4p9.png",
+	"thumbnail": "https://i.imgur.com/U4DlHTe.png",
+	"color": "#41AAA5",
+	"category": "other",
+	"tags": ["livelinkbio", "live", "link", "bio"],
```
Comment (ImRodry) [inline websites/L/LiveLinkBio/dist/metadata.json:18]: maybe consider adding some more like url and shortener?

File: websites/L/LiveLinkBio/presence.ts [hunk 17]
```diff
@@ -0,0 +1,78 @@
+const presence = new Presence({
+		clientId: "966711989533544580"
+	}),
+	browsingStamp = Math.floor(Date.now() / 1000);
```
Comment (ImRodry) [inline websites/L/LiveLinkBio/presence.ts:4]:
```suggestion
	browsingTimestamp = Math.round(Date.now() / 1000);
```

File: websites/L/LiveLinkBio/presence.ts [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+		);
+	let title: string;
+	if (privacy) presenceData.details = "Browsing...";
+	else if (document.location.hostname.includes("docs.livelinkbio.com")) {
+		presenceData.startTimestamp = browsingStamp;
+		if (search && !search.value) {
+			presenceData.details = "Searching For:";
+			presenceData.state = search.value;
+		} else {
+			presenceData.details = "Reading Docs:";
```
Comment (ImRodry) [inline websites/L/LiveLinkBio/presence.ts:25]:
```suggestion
			presenceData.details = "Reading docs:";
```

File: websites/L/LiveLinkBio/presence.ts [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+			presenceData.details = "Affiliate";
+			title = "Affiliate";
+		} else if (document.location.pathname.includes("dashboard")) {
+			presenceData.details = "Dashboard";
+			title = "Dashboard";
+		} else if (document.location.pathname.includes("link")) {
+			presenceData.state = document.querySelector("#link_url").textContent;
+			if (document.location.pathname.endsWith("statistics"))
+				presenceData.details = "Statistics of Link:";
+			else presenceData.details = "Editing Link:";
```
Comment (ImRodry) [inline websites/L/LiveLinkBio/presence.ts:49]:
```suggestion
				presenceData.details = "Statistics of link:";
			else presenceData.details = "Editing link:";
```

**Review by ImRodry — CHANGES_REQUESTED** [review]
Also please address Rhys's reviews, some were resolved without actually being resolved

File: websites/L/LiveLinkBio/presence.ts [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+				"body > div.gitbook-root > div > div > div.css-1dbjc4n.r-1ro0kt6.r-16y2uox.r-1wbh5a2 > div.css-1dbjc4n.r-14lw9ot.r-13awgt0 > div > div > div.css-1dbjc4n.r-1ro0kt6.r-18u37iz.r-16y2uox.r-1wbh5a2 > div > div > div.css-1dbjc4n.r-eqz5dr.r-1ifxtd0.r-1ygmrgt > div.css-1dbjc4n.r-18u37iz > div.css-1dbjc4n.r-1ro0kt6.r-16y2uox.r-1wbh5a2 > div"
+			).textContent;
+		}
+	} else if (document.location.hostname.includes("status.livelinkbio.com"))
+		presenceData.details = "Status";
+	else {
+		const type = document.querySelector(
+				"body > main > section > div.row.mb-4 > div.col-12.col-lg.d-flex.align-items-center.mb-3.mb-lg-0 > h1"
+			),
+			active = document.querySelectorAll('[class="nav-link active"]')[0];
```
Comment (ImRodry) [inline websites/L/LiveLinkBio/presence.ts:36]: pretty sure this does the same job but please test this before committing
```suggestion
			active = document.querySelector('[class="nav-link active"]');
```

File: websites/L/LiveLinkBio/presence.ts (hunk 14 above)
Comment (ImRodry) [reply websites/L/LiveLinkBio/presence.ts:69]: Please address this, don't resolve conversations without addressing them.

File: websites/L/LiveLinkBio/presence.ts (hunk 12 above)
Comment (ImRodry) [reply websites/L/LiveLinkBio/presence.ts:22]: This wasn't addressed.

**Review by ImRodry — CHANGES_REQUESTED** [review]

File: websites/L/LiveLinkBio/presence.ts [hunk 21]
```diff
@@ -0,0 +1,90 @@
...
+					url: document.location.href
+				}
+			];
+			presenceData.details = `All ${title}`;
+		} else if (active) {
+			title = active.textContent.trim();
+			presenceData.details = title;
+		}
+	}
+	if (privacy || !buttons) delete presenceData.buttons;
```
Comment (Slowlife01) [inline websites/L/LiveLinkBio/presence.ts:86]:
```suggestion
	if (!buttons) delete presenceData.buttons;
```
It only show `browsing` so

File: websites/L/LiveLinkBio/presence.ts [hunk 22]
```diff
@@ -0,0 +1,90 @@
...
+		]),
+		search = document.querySelector<HTMLInputElement>(
+			"#portals-root > div:nth-child(3) > div > div > div > div > div.css-1dbjc4n.r-1ro0kt6.r-16y2uox.r-1wbh5a2 > input"
+		);
+	let title: string;
+	if (privacy) presenceData.details = "Browsing...";
+	else if (document.location.hostname.includes("docs.livelinkbio.com")) {
+		if (search?.value) {
+			presenceData.details = "Searching for:";
+			presenceData.state = search.value ?? "Unknown";
```
Comment (ImRodry) [inline websites/L/LiveLinkBio/presence.ts:23]: you check if search.value is truthy above so no need to have a nullish check here
```suggestion
			presenceData.state = search.value;
```

**Review by ImRodry — CHANGES_REQUESTED** [review]
Small nit, otherwise LGTM

**Review by ImRodry — APPROVED** [review]

File: websites/L/LiveLinkBio/dist/metadata.json (hunk 11 above)
Comment (QkeleQ10) [reply websites/L/LiveLinkBio/dist/metadata.json]: The Dutch translation should ideally be `LiveLinkBio is een geavanceerd URL-verkortingsplatform waar je makkelijk links kan verkorten. Bereik je doelgroep op basis van locatie of platform en ontvang inzichten voor je verkorte links.`

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**